local.tf

```diff
@@ -0,0 +1,566 @@
+
+  # Ensure that IAM Role permissions are created before and deleted after EKS Node Group handling.
+  # Otherwise, EKS will not be able to properly delete EC2 Instances and Elastic Network Interfaces.
+  depends_on = [
+    aws_iam_role_policy_attachment.wandb_node_worker_policy,
+    aws_iam_role_policy_attachment.wandb_node_cni_policy,
+    aws_iam_role_policy_attachment.wandb_node_registry_policy,
+  ]
+}
+
+##########################################
+# Load Balancing
+##########################################
+
+resource "aws_security_group" "wandb_alb" {
+  name        = "wandb-alb-sg"
+  description = "Allow http(s) traffic to wandb"
+  vpc_id      = aws_vpc.wandb.id
+
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+    ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "wandb-alb"
+  }
+}
+
+resource "aws_lb" "wandb" {
+  name               = "wandb-alb"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.wandb_alb.id]
+  subnets            = aws_subnet.wandb.*.id
+}
+
+output "lb_address" {
+  value = "http://${aws_lb.wandb.dns_name}"
+}
+
+resource "aws_lb_target_group" "wandb_tg" {
+  name     = "wandb-alb-tg"
+  port     = 32543
+  protocol = "HTTP"
+  vpc_id   = aws_vpc.wandb.id
+
+  health_check {
+    protocol            = "HTTP"
+    path                = "/healthz"
+    port                = "traffic-port"
+    healthy_threshold   = 5
+    unhealthy_threshold = 2
+    matcher             = "200"
+  }
+}
+
+resource "aws_lb_listener" "wandb_listener" {
+  load_balancer_arn = aws_lb.wandb.arn
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.wandb_tg.arn
+  }
+}
+
+resource "aws_autoscaling_attachment" "wandb" {
+  autoscaling_group_name = aws_eks_node_group.eks_worker_node_group.resources[0].autoscaling_groups[0].name
+  alb_target_group_arn   = aws_lb_target_group.wandb_tg.arn
+}
+
+##########################################
+# SQS/SNS
+##########################################
+
+resource "aws_sqs_queue" "file_metadata" {
+  name = "wandb-file-metadata"
+
+  # enable long-polling
+  receive_wait_time_seconds = 10
+}
+
+resource "aws_sqs_queue_policy" "file_metadata_queue_policy" {
+  queue_url = aws_sqs_queue.file_metadata.id
+
+  policy = data.aws_iam_policy_document.file_metadata_queue_policy.json
+}
+
+data "aws_iam_policy_document" "file_metadata_queue_policy" {
+  statement {
+    actions   = ["SQS:SendMessage"]
+    effect    = "Allow"
+    resources = [aws_sqs_queue.file_metadata.arn]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
```

Comment:
This should just be the role the pod uses and the S3 bucket we create if we're following least privilege principles...

Reply:
My policy game is weak, but I think the `condition` below handles locking this down such that only the sns topic can push messages. I looked at the docs for the `principal` block, and it doesn't look like a specific IAM resources (like the sns topic) can be specified here. Still...it's probably worth putting the current account as the principal in here? What do you think?

Reply:
Yeah, current account sounds good.

Reply:
Fixed now.